main.go

```diff
@@ -72,6 +85,37 @@ func createPrincipalName(userArn arn.ARN) (string, error) {
 	}
 }
 
+func (c *config) validatePublicKey(sshPubKey ssh.PublicKey) error {
+	_, ok := sshPubKey.(ssh.CryptoPublicKey)
+	if !ok {
+		return fmt.Errorf("hallow: ssh public key is not a CryptoPublicKey")
+	}
+
+	pubKey := sshPubKey.(ssh.CryptoPublicKey).CryptoPublicKey()
+
+	switch pubKey.(type) {
+	case *rsa.PublicKey:
+		// rsa.PublicKey.Size() returns *bytes*. In an effort to make it easy
+		// to manage this list without doing math to figure out what the switch
+		// cases are, we're going to multiply this by 8 to get to *bits*, which
+		// is the number we all know and love.
+		size := (pubKey.(*rsa.PublicKey).Size() * 8)
```

Comment:
FWIW you can use `.N.BitLen()` if you want to skip this math.

Reply:
A good point!